### DISCUSSION TITLE
Is there a way to use scikit-survival in pycaret?

### DISCUSSION THREAD

**Discussion by loneblues** [pycaret/pycaret#2564] [Q&A] [answered]
Scikit-survival is compatible with scikit-learn, but due to the specific data formats that are required in survival analysis, namely the structure of 'y' appears something like below, it is quite difficult to make it work in pycaret. I didn't figure out whether the specific models such as CoxRegression models, or such used in survival analysis works as a custom model in pycaret, since it was not possible to setup the configurations in the first place. Is there any solution?  
![image](https://user-images.githubusercontent.com/21212266/169029044-a684a353-241a-4c0a-bb92-4853787231a4.png)
Source: https://scikit-survival.readthedocs.io/en/stable/index.html#

Thanks.

**Comment by moezali1** [comment, marked as answer]
@loneblues Unfortunately not. It is out of scope for PyCaret at the moment.

Thanks.

**Comment by loneblues** [comment]
Thanks for the answer!

**Comment by ngupta23** [comment]
Survival Analysis is a fairly popular task. This might be a good addition to PyCaret. As far as I know, none of our current core contributors have experience with this though. If anyone from the community wants to own this module and take the initiative to add this to PyCaret, we would be open to that.

**Reply by moezali1** [reply]
Yes I love the idea. @reza1615 has worked on Survival Analysis. He knows a fair bit.

**Reply by ngupta23** [reply]
Awesome. I was not aware. This would be a nice addition.
